File: Frontend/refinery-management-frontend/src/services/stationService.ts
import { API } from './api';

const getStations = async () => {
    const response = await API.get('http://localhost:8001/api/stations/');
    return response.data;
};

const getStationStocks = async (stationId: number) => {
    const response = await API.get(`http://localhost:8001/api/stations/${stationId}/station-stocks/`);
    return response.data;
};

export default {
    getStations,
    getStationStocks
};

File: Frontend/refinery_driver_app/lib/services/station_service.dart
// station_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import 'auth_service.dart';

class StationService {
  static const String baseUrl = 'http://localhost:8001';

  Future<List<dynamic>> getStations() async {
    final token = await AuthService().getToken();
    final response = await http.get(
      Uri.parse('$baseUrl/api/stations/'),
      headers: {'Authorization': 'Bearer $token'},
    );
    if (response.statusCode == 200) {
      return jsonDecode(response.body);
    } else {
      throw Exception('Failed to load stations');
    }
  }

  Future<dynamic> getStationStocks(int stationId) async {
    final token = await AuthService().getToken();
    final response = await http.get(
      Uri.parse('$baseUrl/api/stations/$stationId/station-stocks/'),
      headers: {'Authorization': 'Bearer $token'},
    );
    if (response.statusCode == 200) {
      return jsonDecode(response.body);
    } else {
      throw Exception('Failed to load station stocks');
    }
  }
}
